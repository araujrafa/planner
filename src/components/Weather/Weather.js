import React from 'react';

export default ({weather, toggleCheckboxChange}) => (
  <div className="checkbox">
    {weather.map(wth => {
      return (
        <label key={wth.id}>
          <input
            type="checkbox"
            value={wth.name}
            onChange={toggleCheckboxChange}
            />
          {wth.name}
        </label>
      )
    })}
  </div>
);
