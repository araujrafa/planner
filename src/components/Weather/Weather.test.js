import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Weather from './Weather';

describe('Weather component', () => {
  const props = {
    weather: [{name: 'Cold'}, {name: 'Hot'}],
    toggleCheckboxChange: jest.fn(),
    label: 'Resultado'
  }

  test('Weather changes the class when hovered', () => {  
    const component = renderer.create(
      <Weather {...props}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Weather call funtion after change', () => {
    const wrapper = shallow(<Weather {...props}/>);
  
    wrapper.find('#Cold').simulate('change');
  
    expect(props.toggleCheckboxChange).toHaveBeenCalledTimes(1);
  });
})