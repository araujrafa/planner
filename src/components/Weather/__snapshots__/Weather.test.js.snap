// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component Weather changes the class when hovered 1`] = `
<div>
  <label>
    Resultado
  </label>
  <div
    className="form-grup"
  >
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        id="Cold"
        onChange={[Function]}
        type="checkbox"
        value="Cold"
      />
      <label
        className="form-check-label"
        htmlFor="Cold"
      >
        Cold
      </label>
    </div>
    <div
      className="form-check form-check-inline"
    >
      <input
        className="form-check-input"
        id="Hot"
        onChange={[Function]}
        type="checkbox"
        value="Hot"
      />
      <label
        className="form-check-label"
        htmlFor="Hot"
      >
        Hot
      </label>
    </div>
  </div>
</div>
`;
